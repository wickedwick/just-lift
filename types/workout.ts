export type Exercise = {
  id: string,
  name: string
  sets: number
  reps: number
  weight: number
  weightUnit: WeightUnit
  progressiveOverload: boolean
  overloadIncrement: number
}

export type Workout = {
  id: string
  exercises: Exercise[]
}

export type WorkoutPlan = {
  workouts: Workout[]
  daysPerWeek: number
  workoutIndex: number
  workoutInProgress: boolean
}

export type Log = {
  date: Date
  workoutId: string
  data: string
  exerciseName: string
  weight: number
  weightUnit: WeightUnit
}

export enum WeightUnit {
  Lbs,
  Kg
}
 