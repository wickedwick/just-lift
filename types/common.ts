import { Exercise, Log, WorkoutPlan } from './workout';
import { StyleProp, TextStyle, ViewStyle } from 'react-native';

export type RootStackParamList = {
  Root: undefined,
  NotFound: undefined
}

export type BottomTabParamList = {
  AutoTagger: undefined
  Celebrities: undefined
}

export type TabOneParamList = {
  BlankMenuScreen: undefined
  CreateWorkoutScreen: undefined
  SelectWorkoutScreen: undefined
  WorkoutScreen: undefined
  SettingsScreen: undefined
  ProfileScreen: undefined
  GuidedSelectionScreen: undefined
  WorkoutListScreen: undefined
}

export type TabTwoParamList = {
  TabOneScreen: undefined
  TabTwoScreen: undefined
}

export type WorkoutFormProps = {
  exercise?: Exercise
  onSubmit: (exercise: Exercise) => void
  onCancel: () => void
}

export type ExerciseCardProps = {
  exercise: Exercise
  onRemovePress: () => void
  onEditPress: () => void
}

export type ExerciseCounterProps = {
  exercise: Exercise
  setLogData: (log: Log) => void
  setWeight: (weight: number, exerciseName: string) => void
}

export type LogCardProps = {
  log: Log
}

export type ActionButtonProps = {
  onPress: () => void
  contained?: boolean
  icon?: string
  labelStyle?: StyleProp<TextStyle>
  style?: StyleProp<ViewStyle>
  text: string
}

export type RadioButtonProps = {
  label: string
  value: string
}

export type CheckboxProps = {
  label: string
  status: 'checked' | 'unchecked'
  onPress: () => void
}

export type WorkoutPlanContextProps = {
  workoutPlan: WorkoutPlan | null
  setWorkoutPlan: (workoutPlan: WorkoutPlan | null) => void
}
