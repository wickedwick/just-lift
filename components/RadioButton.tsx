import React from 'react';
import { RadioButton as PaperRadioButton, useTheme } from 'react-native-paper';
import { RadioButtonProps } from '../types/common';
import { StyleSheet, View } from 'react-native';

const RadioButton = (props: RadioButtonProps): JSX.Element => {
  const { colors } = useTheme()
  
  return (
    <View style={styles.radioContainer}>
      <View style={styles.radio}>
        <PaperRadioButton.Item value={props.value} color={colors.primary} label={props.label} />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  radio: {
    borderColor: '#eee',
    borderWidth: 1,
    borderRadius: 50,
    padding: 5,
  },
  radioContainer: {
    marginBottom: 10,
    width: '20%',
    flex: 1,
    padding: 10,
  },
})

export default RadioButton
