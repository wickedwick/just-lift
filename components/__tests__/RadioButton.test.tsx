import Adapter from 'enzyme-adapter-react-16';
import RadioButton from '../RadioButton';
import React from 'react';
import { configure, shallow } from 'enzyme';
import { RadioButton as PaperRadioButton } from 'react-native-paper';
import { RadioButtonProps } from '../../types/common';

configure({ adapter: new Adapter() })

const props: RadioButtonProps = {
  label: 'New Button',
  value: 'new',
}

describe('<RadioButton />', () => {
  it('Renders', () => {
    const wrapper = shallow(<RadioButton {...props} />)
    const item = wrapper.find(PaperRadioButton.Item)
    
    expect(item).toHaveLength(1)
    expect(item.prop('label')).toEqual('New Button')
    expect(wrapper).toMatchSnapshot()
  })
})
