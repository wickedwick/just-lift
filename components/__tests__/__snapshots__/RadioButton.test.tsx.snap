// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> Renders 1`] = `ShallowWrapper {}`;
